client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -108,221 +59,125 @@ final case class Client[F[_]](
     * [[toKleisli]], and [[streaming]] are safer alternatives, as their
     * signatures guarantee disposal of the HTTP connection.
     */
-  def toHttpService: HttpService[F] =
-    open
-      .map {
-        case DisposableResponse(response, dispose) =>
-          response.copy(body = response.body.onFinalize(dispose))
+    def toHttpService: HttpService[F] = 
+      Kleisli{req: Request[F] => OptionT.liftF(fetch(req)(_.pure[F]))}
```

Comment:
Alright this I find interesting: If reading the body of the request after fetch results in an error, I think `toHttpService` might not be expressible in terms of `fetch`

Reply:
It's more problematic than that: the response is going to be closed when the `_.pure[F]` runs, so we're returning a dead response body.  We can't go from `Stream` to `F` without closing the resource.